src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs

```diff
@@ -16,46 +16,76 @@
 
 using System.Diagnostics.Metrics;
 using System.Reflection;
+using System.Threading;
 using Diagnostics = System.Diagnostics;
 
 namespace OpenTelemetry.Instrumentation.Process;
 
-internal class ProcessMetrics
+internal sealed class ProcessMetrics
 {
     internal static readonly AssemblyName AssemblyName = typeof(ProcessMetrics).Assembly.GetName();
     internal static readonly Meter MeterInstance = new(AssemblyName.Name, AssemblyName.Version.ToString());
-    private static readonly Diagnostics.Process CurrentProcess = Diagnostics.Process.GetCurrentProcess();
 
-    static ProcessMetrics()
+    private static readonly ThreadLocal<InstrumentsValues> CurrentThreadInstrumentsValues = new(() => new InstrumentsValues());
+
+    public ProcessMetrics(ProcessInstrumentationOptions options)
     {
         // TODO: change to ObservableUpDownCounter
         MeterInstance.CreateObservableGauge(
             "process.memory.usage",
-            () =>
-            {
-                CurrentProcess.Refresh();
-                return CurrentProcess.WorkingSet64;
-            },
+            () => CurrentThreadInstrumentsValues.Value.GetMemoryUsage(),
```

Comment:
How does this work with multiple `MeterProvider` setup?

In the scenario where you have multiple meterProviders and all of them add Process instrumentation, this would create the same set of instruments for each `MeterProvider` since they are all using the same `Meter` instance.

Reply:
I am not sure for what intent would the user want to do `.AddProcessInstrumentation()` for multiple meterProviders to a given application/process. Everything in ProcessMetrics should ideally be static in the scope of a Process.
 
Could you share with me the use case you have in mind that ProcessMetrics was being added to more than one meterProviders?

Reply:
One use case could be where the user wants to send data from different meters to different exporters along with the process metrics.

Reply:
I think there will not be a problem if we mark the ctor of ProcessMetrics as static.
I still don't quite get why would the user want to monitor a static object via different exporters?
Once case I could think of now is role-based access control - authorized users of the application can see a certain field like for example process.disk.io while non-authorized users cannot?

Reply:
Updated to use instance members for meter and InstrumentValues with this commit 13434f599fcfc618d00ad5bb4a3872f5823c5c95.

One caveat with this implementation is in `Snapshot()`, threadA calls Process.Refresh(), and before threadA does the assignment to this.memoryUsage, threadB calls Process.Refresh(). The odds are small as assignments should be very fast.
Even if we hit this situation, it should not be a big deal because the time difference between these 2 refreshes is tiny.

Reply:
Attached the discussion I had with Blanch and Utkarsh below.
[discussion.txt](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/files/9711796/discussion.txt)

Reply:
> One caveat with this implementation is in `Snapshot()`, threadA calls Process.Refresh(), and before threadA does the assignment to this.memoryUsage, threadB calls Process.Refresh(). The odds are small as assignments should be very fast. Even if we hit this situation, it should not be a big deal because the time difference between these 2 refreshes is tiny.

When would this happen? Is there a concrete example?

Reply:
![image](https://user-images.githubusercontent.com/16979322/194122587-9d1a6159-7cc3-47f6-adda-888fb9a2a081.png)

The case I have in mind is:
At t3, the value got assigned to `this.virtualMemoryUsage` for threadA is from this.currentProcessRefresh() from t2 by threadB; 
because [GetCurrentProcess()](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.process.getcurrentprocess?view=net-7.0) is a static method that returns "A new [Process](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.process?view=net-7.0) component associated with the process resource that **is** running the calling application." 

Even though we maintain different copies for currentProcess here:
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/13434f599fcfc618d00ad5bb4a3872f5823c5c95/src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs#L51 
Those are referencing the same object.

But again, I don't think this is a case that could be hit frequently. And even it did occur, the time difference between the 2 snapshots are negligible.

Reply:
yeah.... `Snapshot()` will not be accessed concurrently because it is an instance method.
What I was worried about is that whether those 2 instances are refreshing on the same object.

After further investigation, that is not the case:
![image](https://user-images.githubusercontent.com/16979322/194155593-e62496cd-e843-42bd-9c26-4f321d21eeef.png)

Reply:
> How does this work with multiple `MeterProvider` setup?
> 
> In the scenario where you have multiple meterProviders and all of them add Process instrumentation, this would create the same set of instruments for each `MeterProvider` since they are all using the same `Meter` instance.

Please take a look:
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/664/files#r989475625